crates/ruff_linter/src/rules/ruff/rules/unformatted_special_comment.rs

```diff
@@ -0,0 +1,553 @@
+use crate::Locator;
+use regex::Regex;
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_trivia::CommentRanges;
+use ruff_text_size::TextRange;
+use ruff_text_size::TextSize;
+use std::ops::Range;
+use std::sync::LazyLock;
+
+type HintRelativeRange = Range<usize>;
+type SpecialCommentDescriptor<'a> = Option<(HintRelativeRange, &'a str, SpecialComment)>;
+
+#[derive(Debug, Eq, PartialEq)]
+enum SpecialComment {
+    /// `# noqa`, `# noqa: A123, B456`
+    Noqa(Option<Vec<String>>),
+    /// `# ruff: noqa`, `# ruff: noqa: A123, B456`,
+    /// `# flake8: noqa`, `# flake8: noqa: A123, B456`
+    FileLevelNoqa {
+        hint: String,
+        codes: Option<Vec<String>>,
+    },
+
+    /// `# fmt: on`, `# fmt: off`, `fmt: skip`
+    Fmt(String),
+    /// `# isort: skip_file`
+    Isort(String),
+    /// `# mypy: ignore-errors`
+    Mypy(String),
+    /// `# nopycln: import`
+    Nopycln(String),
+    /// `# pyright: strict`, `# pyright: ignore[reportFoo]`
+    Pyright(String),
```

Comment:
I'm somewhat hesitant supporting non-ruff suppression comments:
1. All other `RUF1XX` rules are specific to ruff pragma comments
2. I don't see us as the authority defining how `pyright` and `mypy` comments should be formatted.